docs/pages/material-ui/migration/migrating-from-deprecated-APIs.js

```diff
@@ -0,0 +1,7 @@
+import * as React from 'react';
+import MarkdownDocs from 'docs/src/modules/components/MarkdownDocs';
+import * as pageProps from 'docs/data/material/migration/migrating-from-deprecated-apis/migrating-from-deprecated-apis.md?@mui/markdown';
+
+export default function Page() {
+  return <MarkdownDocs {...pageProps} />;
+}
```

Comment:
The file pathname casing is wrong

Actual: `docs/pages/material-ui/migration/migrating-from-deprecated-APIs.js`
Expected: `docs/pages/material-ui/migration/migrating-from-deprecated-apis.js`

it leads to two bugs: 

1. Side nav not expanded:
<img src="https://github.com/mui/material-ui/assets/3165635/47a5af35-56c8-403d-9b3a-949a8424d886" width="512">

2. Console errors:
<img src="https://github.com/mui/material-ui/assets/3165635/13d062f0-6bdb-420a-8720-05310230a9b3" width="317">

Can we fix it? :)